### PR TITLE
Bugfix/test fail

### LINKED ISSUE #6: Developer: Need Checkbox for Boolean Input
While there is a checkbox test that wants options, if you have a single checkbox with no options, tests error with the following code: 

```
plugin › HTML is written correctly, includes required attributes, and is render-able
    ✖  failed with "Cannot read property '1' of null" at /Users/rachelwhite/Repos/input-plugin-boolean/node_modules/punchcard-shared-tests/lib/input-plugins.js:202:30
```

``` gherkin
Feature: Shared test for single boolean checkbox
  As a developer
  I want a checkbox test for boolean operators
  So that users may have a field for true / false needs

  Scenario: I am developing a form that gives a field where the user has to opt-in or out
    Given a single checkbox that signifies a true / false statement
    When tests are run
    Then they should be able to differentiate it from input plugins with multiple checkboxes and options and allow it to pass.
```

---
- [ ] Test for Boolean Checkbox / Boolean Checkboxes dont have options, so tests need to be run accordingly

### PR DESCRIPTION
**Pull request by scottnath**
@ayush2k @Snugug 

Fix for failing boolean.

Utils script needs to be integrated into shared tests:
https://github.com/punchcard-cms/punchcard/issues/202

but this fix is required now because boolean is failing in all implementations

---

Resolves https://github.com/punchcard-cms/shared-tests/issues/6

`DCO 1.1 Signed-off-by: Scott Nath <github@scottnath.com>`

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -6,31 +6,40 @@
  *
  * A simple boolean input type
  */
-const validation = require('./lib/validation.js');
+const clone = require('lodash/clonedeep');
+const inpugCheckbox = require('input-plugin-checkbox');
```
Comment (Snugug) [inline index.js]: probably `inputCheckbox`, but maybe just `checkbox`?

File: index.js [hunk 2]
```diff
@@ -6,31 +6,40 @@
...
 
+const utils = require('./lib/utils');
+
+// clone plugins for each input
+const boolean = clone(inpugCheckbox);
+
+// make replacements in html
+boolean.html = boolean.html.replace(/checkbox\./g, 'boolean.');
```
Comment (Snugug) [inline index.js]: Not going to want to do this. `input type="checkbox"` will be replaced with `input type="boolean"` and that's not what we want

File: lib/utils.js [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+            v[val] = plugin.validation[val];
+          }
+        });
+      }
+    });
+  }
+
+  return v;
+}
+
```
Comment (Snugug) [inline lib/utils.js]: These functions are identical except for script/validation. Can they be one function and pass in `key` that would be `'validation'` or `'script'` and return the proper thing from there?

You can create helpers that automatically pass in the correct string if you'd like

**Comment by Snugug** [comment]
@scottnath Linting issues are causing tests to fail

File: index.js (hunk 2 above)
Comment (scottnath) [inline index.js]: Not in this case. Regex is looking for `checkbox.`

File: index.js (hunk 1 above)
Comment (scottnath) [inline index.js]: Totally
Agreed

File: index.js [hunk 4]
```diff
@@ -6,31 +6,40 @@
...
  * Single Boolean Input Plugin
  * @module booleanInputPlugin
  */
 module.exports = {
   name: 'Boolean',
   description: 'A simple boolean input type',
-  validation: {
-    booleanValidation: validation,
-  },
+  validation,
```
Comment (ayush2k) [inline index.js]: I think it should be `validation: validation`

Comment (scottnath) [inline index.js]: Our linting only allows for the single key (means the same without the value)

**Comment by scottnath** [comment]
@Snugug @ayush2k 

That can go down my _worst PR of 2016_™

Re-did it, removed the js, and made it simple...which it is. Please re-review the new.

File: index.js [hunk 5]
```diff
@@ -13,24 +13,28 @@ const validation = require('./lib/validation.js');
...
         function: 'booleanValidation',
-        on: 'blur',
+        on: 'change',
       },
-      label: 'A true or false statement goes here',
-      placeholder: 'Boolean',
       type: 'checkbox',
+      label: 'Select from the following',
+      options: [
+        { label: 'true',
```
Comment (Snugug) [inline index.js]: That looks funky. Shouldn't `label` be on the next line?

File: index.js (hunk 2 above)
Comment (Snugug) [inline index.js]: Up, yup, forgot type was compiled in

**Comment by Snugug** [comment]
Only comment: the default validation of "can't be left blank" isn't needed. The validation for this should either be validating that it's true/false (or 1/0) or just, no validation, to return `true`, right?

**Comment by Snugug** [comment]
Also, this doesn't appear to have Coveralls set up (greenkeeper, though, apparently, as there are 51 open PRs). Do you want to cover that in this PR?

Finally, need DCO signoff

**Comment by Snugug** [comment]
By Coveralls not set up, I mean it _is_, but the repo isn't set up to send to Coveralls

**Comment by scottnath** [comment]
@Snugug I think this plugin must have been missed in the sweep to upgrade all the plugins (thus the missing template for PRs as well.) Let's cover that in a different PR. 
I've changed the orignal PR message and some code per your comments as well.

**Comment by scottnath** [comment]
here it is: https://github.com/punchcard-cms/input-plugin-boolean/pull/30

will pick that up once this PR is merged in.

**Comment by Snugug** [comment]
@scottnath Got yo self some linting issues

**Comment by scottnath** [comment]
@Snugug FIXED! ready for Mergy McMergenstein.

**Pull request merged**